pkg/controller/datadogagentdeployment/datadogagentdeployment_controller.go

```diff
@@ -202,6 +220,8 @@ func (r *ReconcileDatadogAgentDeployment) Reconcile(request reconcile.Request) (
 		return reconcile.Result{}, err
 	}
 
+	r.forwarders.Register(request.NamespacedName)
```

Comment:
Can we "Register" the resource only once? 
Maybe by call register only on Add() event in the watcher

Reply:
makes perfect sense 👍